### ISSUE TITLE
No contacts shown when logged in as admin after update to version 3.0.0

### ISSUE THREAD

**Issue by phiL0co** [nextcloud/contacts#774]
### Steps to reproduce
1. Upgrade contacts to version 3.0.0
2. Log into nextcloud as admin 
3. open contacts app

### Expected behaviour
- The contacts should be visible as before
- It should be possible to create a new contact

### Actual behaviour
- contact list is empty
- it is not possible to add a new contact
- contacts still appear when searching contacts in the upper right corner

### Server configuration

**Operating system**:
Ubuntu 16.04

**Web server:**
Apache

**Database:**
mysql 5.7.24

**PHP version:**
PHP 7.0.32-0ubuntu0.16.04.1

**Nextcloud version:** (see Nextcloud admin page)
15

**Contacts version:** (see Nextcloud apps page)
3.0.0

**Updated from an older Nextcloud or fresh install:**
updated

**Signing status:**

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
No errors have been found.

**List of activated apps:**

 Accessibility 1.1.0
Activity 2.8.2
Auditing / Logging 1.5.0
Bookmarks 0.14.3
Calendar 1.6.4
Collaborative tags 1.5.0
Comments1.5.0
Contacts 3.0.0
Default encryption module 2.3.0
Deleted files 1.5.0
Federation 1.5.0
File sharing 1.7.0
First run wizard 2.4.0
Gallery 18.2.0
Log Reader 2.0.0
Markdown Editor 2.0.5
Metadata 0.8.0
Monitoring 1.5.0
Nextcloud announcements 1.4.0
Notifications2.3.0
Password policy 1.5.0
PDF viewer 1.4.0
RainLoop 6.0.2
Share by mail 1.5.0
Support 1.0.0
Tasks 0.9.8
Text editor 2.7.0
Theming 1.6.0
Update notification 1.5.0
Usage survey 1.3.0
Versions 1.8.0
Video player 

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your instance's installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```

I have no access to the config at the moment.

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
yes

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Firefox

**Operating system:**
Arch Linux / Windows

#### Web server error log
```

```
no related errors shown

**Comment by skjnldsv** [comment]
Hello!
I will need your browser console log to investigate this issue.
Open your console, reload your page and/or do the action leading to this issue and copy/paste the log in this thread.

<details>
<summary>How to access your browser console (Click to expand)</summary>

# Chrome
- Press either CTRL + SHIFT + J to open the “console” tab of the Developer Tools.
- Alternative method:
    1. Press either CTRL + SHIFT + I or F12 to open the Developer Tools.
    2. Click the “console” tab.

# Safari
- Press CMD + ALT + I to open the Web Inspector.
- See Chrome’s step 2. (Chrome and Safari have pretty much identical dev tools.)

# IE9
1. Press F12 to open the developer tools.
2. Click the “console” tab.

# Firefox
- Press CTRL + SHIFT + K to open the Web console (COMMAND + SHIFT + K on Macs).
- or, if Firebug is installed (recommended):
    1. Press F12 to open Firebug.
    2. Click on the “console” tab.

# Opera
1. Press CTRL + SHIFT + I to open Dragonfly.
2. Click on the “console” tab.
</details>

**Comment by phiL0co** [comment]
Hello skjnldsv!

OK, here's the console log:
I had to anonymize an email-adress of one of my contacts (after missing parameter) that appeared in the log.

> JQMIGRATE: Migrate is installed, version 1.4.0
core.js:7:542
@babel/polyfill is loaded more than once on this page. This is probably not desirable/intended and may have consequences if different versions of the polyfills are applied sequentially. If you do need to load the polyfill more than once, use @babel/polyfill/noConflict instead to bypass the warning.
contacts.js:38:51118
New search handler registered
search.js:67:4
Connected to dav! 
Object { rootUrl: "https://nextcloud.philo-co.de/remote.php/dav/", advertisedFeatures: (17) […], currentUserPrincipal: {…}, principalCollections: (4) […], calendarHomes: [], addressBookHomes: (1) […], parser: {…}, _isConnected: true, _request: {…} }
Contacts.vue:237
Object { message: "Missing parameter value in 'EMAIL;TYPE=INTERNET\\,HOME\\\\\\:FIRSTNAME@LASTNAME-xyz.de'", name: "ParserError", stack: "ICAL.parse</i._handleContentLine@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:79304\ni/<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:78606\nICAL.parse</i._eachLine@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:82137\ni@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:78578\nt@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:289:211\nt/</</n<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:380:4865\nt/</<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:380:4841\nI/</<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:38:27686\nI/<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:38:27554\nc@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:38:35413\n" }

**Comment by phiL0co** [comment]
Fyi:

The contact already existed before I updated to 3.0.0.

How can I export the contact when the list does not load?

**Comment by skjnldsv** [comment]
@phiL0co Yes, the contacts app itself doesn't change your contacts unless you do it so. Your contact might have been uploaded on an old nextcloud version that had still not the proper checks? Or it could have been synced by an outside app (android, iPhone...)

You should have an url loading on your browser's network log. Like so:
![capture d ecran_2018-12-17_10-07-34](https://user-images.githubusercontent.com/14975046/50077136-cffa4480-01e3-11e9-8b1e-db9114e701ac.png)


Right click, and open in a new tab. It should trigger a download directly :)

**Comment by phiL0co** [comment]
OK, I changed the Home adress field on my phone, it contained a word with german "Umlaut" which was cut before the Umlaut.
I corrected the adress field and synced the contacts.

Now another contact triggers an error.

Here's the new output:

> JQMIGRATE: Migrate is installed, version 1.4.0
core.js:7:542
@babel/polyfill is loaded more than once on this page. This is probably not desirable/intended and may have consequences if different versions of the polyfills are applied sequentially. If you do need to load the polyfill more than once, use @babel/polyfill/noConflict instead to bypass the warning.
contacts.js:38:51118
New search handler registered
search.js:67:4
Connected to dav! 
Object { rootUrl: "https://nextcloud.philo-co.de/remote.php/dav/", advertisedFeatures: (17) […], currentUserPrincipal: {…}, principalCollections: (4) […], calendarHomes: [], addressBookHomes: (1) […], parser: {…}, _isConnected: true, _request: {…} }
Contacts.vue:237
Object { message: "Missing parameter value in 'EMAIL;TYPE=INTERNET\\,HOME\\\\:FIRSTNAME_LASTNAME@freenet.de'", name: "ParserError", stack: "ICAL.parse</i._handleContentLine@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:79304\ni/<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:78606\nICAL.parse</i._eachLine@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:82137\ni@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:7:78578\nt@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:289:211\nt/</</n<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:380:4865\nt/</<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:380:4841\nI/</<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:38:27686\nI/<@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:38:27554\nc@https://nextcloud.philo-co.de/apps/contacts/js/contacts.js?v=a2f3ccbe-47:38:35413\n" }
addressbooks.js:365:4

The new "error contact" has no Umlaut in any data field though.

**Comment by phiL0co** [comment]
> 
> 
> @phiL0co Yes, the contacts app itself doesn't change your contacts unless you do it so. Your contact might have been uploaded on an old nextcloud version that had still not the proper checks? Or it could have been synced by an outside app (android, iPhone...)
> 
> You should have an url loading on your browser's network log. Like so:
> ![capture d ecran_2018-12-17_10-07-34](https://user-images.githubusercontent.com/14975046/50077136-cffa4480-01e3-11e9-8b1e-db9114e701ac.png)
> 
> Right click, and open in a new tab. It should trigger a download directly :)

No, sorry, there's only this link:

https://nextcloud.philo-co.de/remote.php/dav/addressbooks/users/XYZ/default/

Sorry...
No link to a direct contact, because I couldn't select any.

**Comment by phiL0co** [comment]
When I change the contacts email-type on my phone, from "other" to "private" or "business", the contacts than seems to work an another one with the same email type then creates the error.
Have tested this with several contacts now.
Seems like this is the problem...

**Comment by skjnldsv** [comment]
So wrong type I guess? What phone do you have?
I'll try to reproduce using your data.

I should add a better message to explain the user that the contact read have failed :thinking:

**Comment by phiL0co** [comment]
> 
> 
> So wrong type I guess? What phone do you have?
Seems so.
But it all worked first the previous contacts version...

Wileyfox Swift with LineageOS 15.1.

> I'll try to reproduce using your data.
> 
> I should add a better message to explain the user that the contact read have failed 🤔
That would be helpful! THX!

**Comment by phiL0co** [comment]
Ok.

At them moment, the only thing I can do is change the corrupt contacts on my phone, than sync the one contact change and than find the next one via the console log etc. etc. 

It's very tedious and I have several hundred contacts.
So if there's any way to automate this, I would be very thankful! ;-)

**Comment by skjnldsv** [comment]
@phiL0co I will provide a fix in the following days to at least still show the rest of the contacts. Failing that hard should not happens. :)

**Comment by phiL0co** [comment]
OK, thx a lot!

**Comment by skjnldsv** [comment]
Fix: https://github.com/nextcloud/contacts/pull/775

**Comment by phiL0co** [comment]
So, I corrected all concerned contacts. They now show up in the browser.

Now, when I open a contact, I see, that all fields that have the type "other" (at least on my phone), show up like this:

![2018-12-17_10h56_37](https://user-images.githubusercontent.com/9380245/50079663-78130c00-01ea-11e9-8a9f-c24f9255828b.png)

![2018-12-17_10h59_35](https://user-images.githubusercontent.com/9380245/50079822-ef48a000-01ea-11e9-9f70-b1395d2932d6.png)

![2018-12-17_10h49_03](https://user-images.githubusercontent.com/9380245/50079829-f374bd80-01ea-11e9-8776-853759b8bf3a.png)

Is it possible to prevent this?

It looks really ugly and not intended...

**Comment by skjnldsv** [comment]
@phiL0co Thos are not valid values.
We usually try to overcome those kind of syntax
- `HOME,VOICE,PREF` is invalid and looks like an old 2.1 vcard that was wrongly migrated. This should be `TEL;TYPE=HOME,VOICE;PREF:xxxxxthevalue` 
- Same for `INTERNET,HOME\,PREF`
- The first one (`PREF;TYPE=HOME`) seems valid, so I don't understand why. Could you give me the full value? It's cut on the screenshot

**Comment by phiL0co** [comment]
I must add something:

Now that all contacts show up, I realized that some contacts disappear when I click on them.
A message like "The contacts does not exist on the server" shows up briefly and the entry disappears.
But it reappears after a reload but disappears again when clicked upon. So it still exists, the error message is incorrect.

The disappearing contacts all have the "other" entry for some or several data. When I replace the "other" type with something else (on my phone) and sync it, the contacts stop disappearing in the browser view.

**Comment by skjnldsv** [comment]
Yes, this is a duplicate of #762 
Basically the contact is having issues (not valid) and we cannot read it properly (because we fetch the full contact and not just some part of it like we initially do)

**Comment by franck-eyraud** [comment]
> let the user manually check those than breaking things

I have the same issue. Is there a way to edit the bad vCards to put the correct format ? That doesn't involve to edit them on the phone, hoping that it would upload a correct contact and not completely corrupt it..

**Comment by skjnldsv** [comment]
@franck-eyraud hey!

Any errors in the logs?

**Comment by franck-eyraud** [comment]
Yes, there is, I could identify what needs to be changed, but I do not know how to apply the change... 

```json
message: "Missing parameter value in 'EMAIL;TYPE=INTERNET\\\\:user@domain.tld'"
```

I can confirm that when downloading the whole address book, 25 contacts out of 1K have these double backslashes that seem wrong.

**Comment by skjnldsv** [comment]
there must be no backslash at all indeed :)

**Comment by franck-eyraud** [comment]
Thanks for the confirmation, but no way to fix it ?

**Comment by skjnldsv** [comment]
either manually with a carddav browser
Or directly in the database

**Comment by franck-eyraud** [comment]
do you have some suggestion of carddav browser ?
About the database, I could maybe do it, but how to identify the row ? Seems a bit risky, since they are also hex encoded it seems...

**Comment by skjnldsv** [comment]
you could also delete the broken vcards and upload new ones?

**Comment by franck-eyraud** [comment]
> you could also delete the broken vcards

Yes, but how ? delete directly from the database ? and how to identify them (their ID) ?

**Comment by skjnldsv** [comment]
You don't need the id, but the filenames.
But I should probably add more debugs to this.
Let me provide you with a debug version.

Then this would be the process:
1. get a list of the vcards paths
2. download each failing contact by accessing the path in your browser
  https://YOUR-DOMAIN/remote.php/dav/addressbooks/users/YOUR-USER/THE-ADDRESSBOOK/YOUR-CONTACT-FILENAME.vcf
2. execute the deletion from your browser console on the contacts app
  ```js
  $.ajax({method: 'DELETE', url: 'https://YOUR-DOMAIN/remote.php/dav/addressbooks/users/YOUR-USER/THE-ADDRESSBOOK/YOUR-CONTACT-FILENAME.vcf'})
  ```
4. edit the downloaded vcards and import them again

**Comment by skjnldsv** [comment]
@franck-eyraud can you try this version please:
[contacts.tar.gz](https://github.com/nextcloud/contacts/files/3559664/contacts.tar.gz)

**Comment by franck-eyraud** [comment]
Thank you for your efforts !

> get a list of the vcards paths

How ?

> can you try this version please:

Interesting, but how to install it... (sorry not a Nextcloud guru, always used the default apps, and ran the upgrade, that's it...). I will anyway not be able to do it immediately, as I need to leave. 

If it can help you improve things, I'm happy to try and to help, so I can try the newere version later.

But for my issue, maybe I found a way to do it : from the full VCF file, I can identify the wrong ones by searching for the `\\:` string. There, I can see the UID field which is the same as the vcard filename (UID.vcf). I can edit the VCF (I need to also remove the UID field). Now you told me about the HTTP DELETE method to delete them, that could help. I have few of them, so I could do that manually.

**Comment by skjnldsv** [comment]
> There, I can see the UID field which is the same as the vcard filename (UID.vcf)

Ah, nice then!
It's not always the case if they've been created with another client than nextcloud

Then make sure you indeed have downloaded (backup) ALL you failed vcards
Then you can follow my procedure above.

**Comment by franck-eyraud** [comment]
Hello,

I didn't install the version you sent, instead I tried to solve the issue as a mere user, without using the fact that I'm administrator of the server.

I could manage to fix the situation, _thanks to your help_, but I wouldn't say that it was easy for any user.

The steps were :
1. identify the files : longest part, by finding the \\ occurences in the full address book export, and getting the UID field, which happens to be the same as VCF for these old entries
2. download all the vcf files
4. remove the bad VCF from database files with DELETE requests
3. Fix the VCF files (some sed command)
5. concatenate them in one file, and upload it through the interface (by keeping the uid field, they keep the same uid, but the new VCF file has a different name)

In order to help in step 1, maybe when the message "loading failed for 25 contacts" appears, the list of vcf files in the console would be of help in order to automatize the process (sending the DELETE requests from command lines requires to extract the full curl command from Firefox, but it can be done)

Another comment, when searching my full address book export, I noticed that at least one contact had a format error in the TEL field, without that it was giving any error `TEL;TYPE=TEL\\;TYPE:+3526..`
So maybe this parsing of bad email could be more permissive, and allow to fix the error from the editor.

Thanks again !
